src/config.rs

```diff
@@ -208,6 +208,50 @@ pub trait Config: Sized + Send + Sync + UnwindSafe + RefUnwindSafe + 'static {
         Ok(())
     }
 
+    /// Called upon client connect (if online mode is enabled) to obtain
+    /// the host address of the session server. Defaults to the official
+    /// mojang session server.
+    ///
+    /// This method is called from within the default implementation of
+    /// [`Self::session_server_url`]. If that method is overridden, this
+    /// method will not be called.
+    ///
+    /// # Default Implementation
+    ///
+    /// The official mojang session server (`sessionserver.mojang.com`)
+    /// is used.
+    fn session_server_host(&self, server: &SharedServer<Self>) -> Cow<'_, str> {
```

Comment:
Don't think it's worth having a second function for this. If you're overriding the session server you probably know what you're doing.